libvips/arithmetic/hough_line.c

```diff
@@ -107,22 +105,26 @@ vips_hough_line_vote(VipsHough *hough, VipsImage *accumulator, int x, int y)
 {
 	VipsHoughLine *hough_line = (VipsHoughLine *) hough;
 	VipsStatistic *statistic = (VipsStatistic *) hough;
-	double xd = (double) x / statistic->ready->Xsize;
-	double yd = (double) y / statistic->ready->Ysize;
+
+	// normalise (x, y) to diagonal size of input image
+	int xs = statistic->ready->Xsize;
+	int ys = statistic->ready->Ysize;
+	double d = sqrt(xs * xs + ys * ys);
+	double xd = (double) x / d;
+	double yd = (double) y / d;
+
+	// size of hough space
 	int width = hough_line->width;
 	int height = hough_line->height;
 	guint *data = (guint *) accumulator->data;
 
-	int i;
-
-	for (i = 0; i < width; i++) {
+	for (int i = 0; i < width; i++) {
 		int i90 = i + width / 2;
 		double r = xd * hough_line->sin[i90] + yd * hough_line->sin[i];
-		int ri = height * r;
+		int ri = r * (height / 2) + height / 2;
```

Comment:
This can be simplified to:
```suggestion
		int ri = (r + 1) * (height / 2.0);
```

Though, I think the compiler already does this.

Reply:
You're right twice, the compiler will do that, and it's clearer.